packages/opentelemetry-resources/src/detect-resources.ts

```diff
@@ -14,103 +14,39 @@
  * limitations under the License.
  */
 
-import { Resource } from './Resource';
-import { ResourceDetectionConfig } from './config';
 import { diag } from '@opentelemetry/api';
-import { isPromiseLike } from './utils';
-import { Detector, DetectorSync } from './types';
 import { IResource } from './IResource';
+import { Resource } from './Resource';
+import { ResourceDetectionConfig } from './config';
 
 /**
- * Runs all resource detectors and returns the results merged into a single Resource. Promise
- * does not resolve until all the underlying detectors have resolved, unlike
- * detectResourcesSync.
+ * Runs all resource detectors and returns the results merged into a single Resource.
  *
- * @deprecated use detectResourcesSync() instead.
  * @param config Configuration for resource detection
  */
-export const detectResources = async (
+export const detectResources = (
   config: ResourceDetectionConfig = {}
-): Promise<IResource> => {
-  const resources: IResource[] = await Promise.all(
-    (config.detectors || []).map(async d => {
-      try {
-        const resource = await d.detect(config);
-        diag.debug(`${d.constructor.name} found resource.`, resource);
-        return resource;
-      } catch (e) {
-        diag.debug(`${d.constructor.name} failed: ${e.message}`);
-        return Resource.empty();
-      }
-    })
-  );
+): IResource => {
+  const resources: Resource[] = (config.detectors || []).map(d => {
+    try {
+      const resource = new Resource(d.detect(config));
+      diag.debug(`${d.constructor.name} found resource.`, resource);
+      return resource;
+    } catch (e) {
+      diag.debug(`${d.constructor.name} failed: ${e.message}`);
+      return new Resource({ attributes: {} });
+    }
+  });
 
   // Future check if verbose logging is enabled issue #1903
   logResources(resources);
 
   return resources.reduce(
     (acc, resource) => acc.merge(resource),
-    Resource.empty()
+    new Resource({ attributes: {} })
```

Comment:
Is there a reason for these changes? It looked to me `Resource` are intended to be/in practice immutable, so it shouldn't matter if these used the shared `Resource.EMPTY` instance, right? (Otherwise – this may apply: https://github.com/open-telemetry/opentelemetry-js/pull/5350/files?diff=unified&w=1#r1925814414)

Reply:
yes i'll switch it to the shared empty instance